evap/staff/views.py

```diff
@@ -1,6 +1,7 @@
 from django.contrib import messages
 from django.core.exceptions import PermissionDenied, SuspiciousOperation
-from django.db.models import Max, Count
+from django.db import transaction, IntegrityError
+from django.db.models import Max, Count, Q
```

Comment:
please check the landscape issue, this is imported somewhere in the file as well. please check whether the imports there are really necessary and if not, could you move them to the top?